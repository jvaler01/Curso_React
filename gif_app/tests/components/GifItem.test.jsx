import {render, screen } from "@testing-library/react";
import {GifItem} from "../../src/components/index.js";

describe('Tests in GifItem', ()=>{
    const title = 'CyberPunk';
    const url = 'https://cyberpunk.com/cyberpunk.jpg'
    test('Snapshot match', ()=>{
        const { container } = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('Should show img with url and alt', ()=>{
        render(<GifItem title={title} url={url}/>);
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe(url);
        expect( alt ).toBe(title);
    });

    test('Should show the title of the component', ()=>{
        render(<GifItem title={title} url={url}/>);
        expect( screen.getByText(title)).toBeTruthy();
    });
})