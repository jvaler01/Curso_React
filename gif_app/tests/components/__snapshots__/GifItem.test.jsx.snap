// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifItem Snapshot match 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="CyberPunk"
      src="https://cyberpunk.com/cyberpunk.jpg"
    />
    <p>
      CyberPunk
    </p>
  </div>
</div>
`;
